onadata/apps/api/viewsets/attachment_viewset.py

```diff
@@ -100,7 +100,9 @@ def list(self, request, *args, **kwargs):
                 if not xform.shared_data:
                     raise Http404(_("Not Found"))
 
-        self.object_list = self.filter_queryset(self.get_queryset())
+        request_data = self.filter_queryset(self.get_queryset())
+        self.object_list = self.object_list = [
```

Comment:
Would filtering this while declaring the `queryset` work ? Like so `Attachment.objects.filter(instance__deleted_at__isnull=True, deleted_at__isnull=True)` ?

Reply:
Aaaaaaah...totally makes sense. I'll change that.